File: server/src/main/java/model/Recurrent.java
package model;

import java.time.LocalDate;

enum TimeType { day, week, month, year; }

class Recurrent {
  private int id;
  private String title;
  private float value;
  private int period;
  private TimeType type;
  private LocalDate startDate;
  private LocalDate endDate;
  private LocalDate updatedAt;
  private LocalDate createdAt; // server-only

  private int walletId;
  private int[] entries;
  private int categoryId; // optional
}

File: client/src/schemas/recurrent.ts
import { z } from "zod";

export const recurringSchema = z.object({
  id: z.number(),
  title: z.string(),
  value: z.number(),
  period: z.number(),
  timeType: z.enum(["day", "week", "month", "year"]),
  startDate: z.number(),
  endDate: z.number(),
  updatedAt: z.number(),

  // Relations
  walletId: z.number(),
  entries: z.array(z.number()),
  categoryId: z.number().optional(),
});

export type Recurrent = z.infer<typeof recurringSchema>;
